fix refactor-nrepl option passthrough

ltriant opened the pull request:
The method of passing through refactor-nrepl options does not work. Instead of attempting to configure via a stringified Clojure map to send to nrepl, allow the user to set options explicitly in a way that is more integrated into the vim ecosystem.

Only supports the `prune-ns-form` and `prefix-rewriting` so far.

This should address #3

ltriant committed:
cider: fix refactor-nrepl option passthrough

The method of passing through refactor-nrepl options does not work.
Instead of attempting to configure via a stringified Clojure map to send
to nrepl, allow the user to set each option explicitly in a way that is
more integrated into the vim ecosystem.

SevereOverfl0w commented:
This appears to include some unrelated changes to keybindings?

SevereOverfl0w commented:
https://github.com/clojure-emacs/refactor-nrepl/commit/3a2fa389485716b427a7758934a1024469a4a7ae#diff-9d54d18d9837001d8284bccf5d4c50b0 looks like this commit changed the behavior.

Is it worth just making this option a map which will be merged into? Then it will support all of the available options.

ltriant commented:
> This appears to include some unrelated changes to keybindings?

That's to get the vim docs inline with what the actual keybindings are. The keybindings were altered some time in the past.

> Is it worth just making this option a map which will be merged into? Then it will support all of the available options.

Good idea! I'll check out how to do that. Vimscript isn't my strongpoint :)

ltriant committed:
cider: configure refactor-nrepl options as a dictionary

Rather than having a global per option, have a single global dictionary
for refactor-nrepl options. This is more versatile and resilient to any
future additions or removals to these options from the refactor-nrepl
project.

ltriant commented:
I've made the options a map. I think that's more versatile now.

Deraen commented on plugin/cider.vim:
```diff
@@ -145,7 +136,15 @@ function! s:clean_ns() abort
+	let opts = { 'op': 'clean-ns', 'path': p }
+
+	if exists('g:refactor_nrepl_options')
+		for [opt_k, opt_v] in items(g:refactor_nrepl_options)
```
I think you can use `extend` function here. Might not even need the `if`.

SevereOverfl0w replied:
You can probably do something like `extend(get(g:,'refactor_nrepl_options', {}), opts)`

Deraen replied:
`extend` modifies the first given dict so you should create local opts dict so the global options dict is not accidentally modified.

```
let opts = {'op': 'clean-ns', 'path': p}
extend(opts, get(g:,'refactor_nrepl_options', {}))
```

ltriant replied:
Ah, thank you. Of course there's an extend function! Much better :)

ltriant committed:
cider: use get() and extend() for better vim-scriptiness

Manually iterating through the items and adding them to the dictionary
is unnecessary as the extend() functions exists to extend a dictionary
with the items of another dictionary.

Deraen commented:
Tested & merged, thanks.
